tests/test_events/test_api.py

```diff
@@ -198,16 +198,36 @@ def test_singleton_pattern_working_correctly_for_event_api(self, mock_send: Magi
             ]
         )
 
-    @patch.object(Boto3Client, "eventbridge", MagicMock())
-    def test_second_send_clears_everything(self) -> None:
+    @patch.object(Boto3Client, "eventbridge")
+    def test_second_send_does_not_clear_everything(self, mock_send: MagicMock) -> None:
```

Comment:
It looks like you mixed up tests with their names 😉 
```suggestion
    def test__send__raises_error_only_when_particular_attempt_failed(
        self, mock_send: MagicMock
    ) -> None:
```